File: app/channels/hits_channel.rb
class HitsChannel < ApplicationCable::Channel
  def subscribed
    # pubsub
    stream_from "hits_channel"
  end

    # the "hits_channel.js" has a "connected" method that will call
    #  the "HitsChannel::hits" method to increment and broadcast 
  def trigger_hits
    puts REDIS.get('hits_count') 
    REDIS.incr('hits_count')
    data = {}
    data['hits_count'] = REDIS.get('hits_count').to_i
    puts data
    ActionCable.server.broadcast('hits_channel', data.as_json)
  end
  
  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
    stop_all_streams
  end
end

File: app/javascript/channels/hits_channel.js
import consumer from "./consumer";

const hitsChannel = consumer.subscriptions.create(
  { channel: "HitsChannel" },
  {
    connected() {
      console.log("hits connected");
      // calling the method "hits" in "hits_channel.rb" and pass an object
      this.perform("trigger_hits");
      this.send({ msg: "new client" });
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received: function (data) {
      // overwritten in Button.jsx
    },
  }
);

export default hitsChannel;
